.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/verify/shared.sh
+source .expeditor/scripts/verify/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+install_rustup
+
+# set minimal profile so only core items + rustfmt are installed
+echo "--- :habicat: Setting minimal profile"
+rustup set profile minimal
+
+for ((i=0; i<MAX_ATTEMPTS; i++)); do
+  # begin searching nightly date versions using current date
+  # as a starting point
+  attempt_date=$(date '+%Y-%m-%d' --date "${TODAY} -${i} day")
+  # use the conversion to seconds for easy date comparison
+  attempt_date_s=$(date '+%s' --date "${attempt_date}")
+  attempt_version="${CHANNEL}-${attempt_date}"
+  echo "--- Trying Nightly Date Version: ${attempt_version}"
+  if rustup toolchain install "${attempt_version}" --component rustfmt; then
+    FOUND_VERSION=${attempt_version}
+    break
+  fi
+  if [ "${attempt_date_s}" -le "${CURRENT_RUSTFMT_DATE_S}" ]; then
+    # Exit early since there is no newer version available than what is
+    # currently pinned in RUSTFMT_VERSION file.
+    echo "--- No newer ${CHANNEL} versions found. Nothing to do."
+    exit 0
+  fi
+done
+
+if [ -z "${FOUND_VERSION}" ]; then
+  # Something is likely wrong if we did not find an installable version.
+  echo "--- :thumbsdown: No candidate versions found within ${MAX_ATTEMPTS}!"
+  exit 1
+fi
+
+# Success on finding a new version
+echo "--- :thumbsup: New installable version of rustfmt found: ${FOUND_VERSION}"
+
+branch="expeditor/bump_rustfmt_${FOUND_VERSION}"
+git checkout -b "${branch}"
+
+# update the version file
+echo "${FOUND_VERSION}" > RUSTFMT_VERSION
+
+# sweep the codebase
+cargo +"$(< RUSTFMT_VERSION)" fmt
+
+# use -u to only add updated files
+git add -u
+git commit -sm "Bump rustfmt to ${FOUND_VERSION}"
+
+# tell expeditor to open a PR
+echo "--- :shipit: Opening a PR"
+open_pull_request
+
+# back to master and cleanup the leftovers
+git checkout -
+git branch -D "${branch}"
+
+read -r -d '' MESSAGE <<'EOF'
+Hi @habitat-team, there's a rustfmt version bump PR ready for review!
+:tada:
```

Comment:
Nice!

Is there any way to grab the URL of the PR that was created so we can add that to the message?

Reply:
I asked @tduffield and there isn't a built-in way.
However, I believe we could leverage the [pull_request](https://expeditor.chef.io/docs/reference/workloads/#pull-request) workload subscription functionality to make this happen. We would probably have to use a standard branch name like `bump_rustfmt` and key off of that to identify the correct PR.

Reply:
I pushed a commit in https://github.com/habitat-sh/habitat/pull/7258/commits/33709bf2983c6d606307a7293616f5caa61f6855 that uses `hub` to allow us to query for the opened PR URL.